app/process/watcher/ethereum/watcher.go

```diff
@@ -101,7 +125,62 @@ func (ew *Watcher) handleLog(eventLog *routerContract.RouterBurn, q *pair.Queue)
 		Id:           fmt.Sprintf("%s-%d", eventLog.Raw.TxHash, eventLog.Raw.Index),
 		Recipient:    recipientAccount,
 		NativeAsset:  nativeAsset,
-		WrappedAsset: eventLog.WrappedAsset.String(),
+		WrappedAsset: eventLog.Token.String(),
+	}
+
+	err = ew.ethClient.WaitForConfirmations(eventLog.Raw)
+	if err != nil {
+		ew.logger.Errorf("[%s] - Failed waiting for confirmation before processing. Error: %s", eventLog.Raw.TxHash, err)
+		return
+	}
+
+	ew.logger.Infof("[%s] - New Burn Event Log with Amount [%s], Receiver Address [%s] has been found.",
+		eventLog.Raw.TxHash.String(),
+		eventLog.Amount.String(),
+		recipientAccount.String())
+
+	q.Push(&queue.Message{Payload: burnEvent})
+}
+
+func (ew *Watcher) handleLockLog(eventLog *router.RouterLock, q qi.Queue) {
+	ew.logger.Debugf("[%s] - New Lock Event Log received. Waiting block confirmations", eventLog.Raw.TxHash)
+
+	if eventLog.Raw.Removed {
+		ew.logger.Errorf("[%s] - Uncle block transaction was removed.", eventLog.Raw.TxHash)
+		return
+	}
+
+	if len(eventLog.Receiver) == 0 {
+		ew.logger.Errorf("[%s] - Empty receiver account.", eventLog.Raw.TxHash)
+		return
+	}
+
+	recipientAccount, err := hedera.AccountIDFromBytes(eventLog.Receiver)
+	if err != nil {
+		ew.logger.Errorf("[%s] - Failed to parse account from bytes [%v]. Error: [%s].", eventLog.Raw.TxHash, eventLog.Receiver, err)
+		return
+	}
+
+	// TODO: Replace with external configuration service
+	wrappedAsset := ew.mappings.NativeToWrappedByNetwork[ew.ethClient.ChainID().Int64()].NativeAssets[eventLog.Token.String()][eventLog.TargetChain.Int64()]
+	if wrappedAsset == "" {
+		ew.logger.Errorf("[%s] - Failed to retrieve native asset of [%s].", eventLog.Raw.TxHash, eventLog.Token)
+		return
+	}
+
+	if wrappedAsset != constants.Hbar && !hederahelper.IsTokenID(wrappedAsset) {
+		ew.logger.Errorf("[%s] - Invalid Native Token [%s].", eventLog.Raw.TxHash, wrappedAsset)
+		return
+	}
```

Comment:
This must be removed when we want to have support for multiple chains not only Hedera 1:1 EVM chain. You can add TODO